### PR TITLE
Add initial notification icons support on Linux and BSD

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

This is a WIP PR for adding the support for showing the application icon in notifications on Linux and BSD. The code needs to be tidied up and the caching needs to be improved so that we only write it once and so on. 
image

It would also be great if we could reuse the application icon if it has been installed into the `/usr/share/icons` path but that might be a bit harder to pull off. Does anyone know an easy way to get the XDG icon corresponding to a specific name?

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [ ] Tests all pass.

### CODE REVIEW

File: app/app_xdg.go [hunk 1]
```diff
@@ -48,6 +48,37 @@ func (a *fyneApp) SendNotification(n *fyne.Notification) {
 	}
 }
 
+func (a *fyneApp) cachedIconPath() string {
+	metadata := a.Metadata()
```
Comment (andydotxyz) [inline app/app_xdg.go:52]: Just use a.Icon(), it first checks settings and then metadata.

File: app/app_xdg.go [hunk 2]
```diff
@@ -48,6 +48,37 @@ func (a *fyneApp) SendNotification(n *fyne.Notification) {
...
+	if metadata.Icon == nil || len(metadata.Icon.Content()) == 0 {
+		return ""
+	}
+
+	dirPath := filepath.Join(rootCacheDir(), a.uniqueID)
```
Comment (andydotxyz) [inline app/app_xdg.go:57]: This should be `a.UniqueID()` which does things when there is none set...

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Nice direction, just a few comments on the uniqueness/icon lookup

**Comment by Jacalz** [comment]
Sorry for the force push I had to rebase on develop to get the latest CI actions.

**Review by andydotxyz — APPROVED** [review]
Pretty cool, thanks :)

**Pull request merged**